app/assets/javascripts/admin/order_cycles/controllers/edit.js.coffee

```diff
@@ -18,5 +19,14 @@ angular.module('admin.orderCycles')
 
     $scope.submit = ($event, destination) ->
       $event.preventDefault()
+      $scope.navClear()
       StatusMessage.display 'progress', t('js.saving')
       OrderCycle.update(destination, $scope.order_cycle_form)
+
+    # Provide a callback for generating warning messages displayed before leaving the page. 
+    orderCycleNavCallback = ->
+      if $scope.order_cycle_form?.$dirty
+        t('admin.unsaved_confirm_leave')
+
+    # Register the NavigationCheck callback
+    NavigationCheck.register(orderCycleNavCallback)
```

Comment:
The name `orderCycleNavCallback` doesn't tell us anything about what it does and you need to explain it in a comment. Maybe a name like `warn_about_unsaved_changes` would be better?